### PR TITLE
test: add test for decimal and pruning for decimal column

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/datasource/file_format/parquet.rs [hunk 1]
```diff
@@ -1023,6 +1023,49 @@ mod tests {
...
+        let exec = get_exec("fixed_length_decimal_legacy.parquet", None, None).await?;
+        let batches = collect(exec, task_ctx.clone()).await?;
+        assert_eq!(1, batches.len());
+        assert_eq!(1, batches[0].num_columns());
+        let column = batches[0].column(0);
+        assert_eq!(&DataType::Decimal(25, 2), column.data_type());
+
+        // parquet use the fixed length binary as the physical type to store decimal
+        // TODO: arrow-rs don't support convert the physical type of binary to decimal
+        // let exec = get_exec("byte_array_decimal.parquet", None, None).await?;
```
Comment (liukun4515) [inline datafusion/core/src/datasource/file_format/parquet.rs:1064]: arrow-rs doesn't support read decimal value from parquet with binary physical type.
why not support this?

Comment (tustvold) [reply datafusion/core/src/datasource/file_format/parquet.rs:1064]: Why would you have a variable length decimal type? What would this mean?

Comment (liukun4515) [reply datafusion/core/src/datasource/file_format/parquet.rs:1064]: @tustvold 
In the parquet of Java version, we can write decimal with the binary or fixed-length binary, so we can get the decimal using the physical type of binary.

we can get the  definition of decimal from the [format spec](https://github.com/apache/parquet-format/blob/54e53e5d7794d383529dd30746378f19a12afd58/src/main/thrift/parquet.thrift#L67)

Decimal can be stored in the parquet as INT32, INT64, FIXED_LEN_BYTE_ARRAY or BYTE_ARRAY.

I can help to implement the BYTE_ARRAY in the arrow-rs.

Comment (liukun4515) [reply datafusion/core/src/datasource/file_format/parquet.rs:1064]: pr https://github.com/apache/arrow-rs/pull/2160 for supporting read decimal from binary physical type in parquet.

File: datafusion/core/src/physical_optimizer/pruning.rs [hunk 2]
```diff
@@ -1418,6 +1452,74 @@ mod tests {
...
+        let expr = col("s1").gt(lit(ScalarValue::Decimal128(Some(500), 23, 2)));
+        let statistics = TestStatistics::new().with(
+            "s1",
+            ContainerStats::new_decimal128(
+                vec![Some(0), Some(400), None, Some(300)], // min
+                vec![Some(500), Some(600), Some(400), None], // max
+                23,
+                2,
+            ),
+        );
```
Comment (mingmwang) [inline datafusion/core/src/physical_optimizer/pruning.rs:1517]: Can you add some large decimal numbers to test the  decimal(23,2) cases?

Comment (liukun4515) [reply datafusion/core/src/physical_optimizer/pruning.rs:1517]: This test case is just used to test the pruning logic.
I will file follow-up pull request to fix the https://github.com/apache/arrow-datafusion/issues/2962 with parquet rowgroup filter/prune.

File: datafusion/core/src/datasource/file_format/parquet.rs [hunk 3]
```diff
@@ -1023,6 +1023,49 @@ mod tests {
...
+
+        let exec = get_exec("fixed_length_decimal_legacy.parquet", None, None).await?;
+        let batches = collect(exec, task_ctx.clone()).await?;
+        assert_eq!(1, batches.len());
+        assert_eq!(1, batches[0].num_columns());
+        let column = batches[0].column(0);
+        assert_eq!(&DataType::Decimal(13, 2), column.data_type());
+
+        // parquet use the fixed length binary as the physical type to store decimal
+        // TODO: arrow-rs don't support convert the physical type of binary to decimal
```
Comment (alamb) [inline datafusion/core/src/datasource/file_format/parquet.rs:1063]:
```suggestion
        // TODO: arrow-rs don't support convert the physical type of binary to decimal
        // https://github.com/apache/arrow-rs/pull/2160
```

**Review by alamb — APPROVED** [review]
LGTM

**Comment by alamb** [comment]
Thanks @liukun4515

**Pull request merged**